# Genome2GenomeDistance

`nucmeer file1 file2 `
`show-coordes out.delta`


#show co-ords output
[S1] start of the alignment region in the reference sequence
 [E1] end of the alignment region in the reference sequence
 [S2] start of the alignment region in the query sequence
 [E2] end of the alignment region in the query sequence
 [LEN 1] length of the alignment region in the reference sequence
 [LEN 2] length of the alignment region in the query sequenc
e
 [% IDY] percent identity of the alignment
 [% SIM] percent similarity of the alignment
 [% STP] percent of stop codons in the alignment
 [LEN R] length of the reference sequence
 [LEN Q] length of the query sequence
 [COV R] percent alignment coverage in the reference sequenc
e
 [COV Q] percent alignment coverage in the query sequence
 [FRM] reading frame for the reference and query sequence alignments respectively
 [TAGS] the reference and query FastA IDs respectively.
 All output coordinates and lengths are relative to the forward strand
 
 #Read This
 http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2612450/
